### ISSUE TITLE
Switching layers results in incorrect annotation

### ISSUE THREAD

**Issue by katiePitchBook** [webanno/webanno#372]
Hi!  Here's a screenshot!  When I switch from one layer to another layer and try to annotate the same text, the selected text does not get annotated.  
  
![image](https://cloud.githubusercontent.com/assets/15332740/19205046/e5ae98ec-8c94-11e6-9fa2-010e93fc9bbb.png)

**Comment by reckart** [comment]
Just tried that (with "Remember layer" disabled in the settings dialog of the action bar):

* select layer POS
* select word
* POS annotation gets created
* press "clear" button
* select layer Lemma
* select same word again
* Lemma annotation gets created

Can you describe how exactly to reproduce your problem?

**Comment by katiePitchBook** [comment]
Yes of course!

- Select Layer Type 1
- Select Word
- Switch from Layer Type 1 to Layer Type 2
- Following Dialog box comes up.  Click Yes.
![image](https://cloud.githubusercontent.com/assets/15332740/19205945/3b8cd256-8c9a-11e6-99a8-688e4b357850.png)

Results:  Layer Type 2 annotation gets created.  Not on the selected text, but prior to the first word.

**Comment by reckart** [comment]
Ok, I see the problem. The new annotation get's created at position 0 with zero width... definitely a bug :)

**Comment by reckart** [comment]
The cause for this issue is #373 - the accidental creation of zero-width Token annotations.

@katiePitchBook due to #373, you now likely have zero-width Token annotations in your CAS. These must be removed. It should be possible to do this via a [new CAS Doctor repair action](https://zoidberg.ukp.informatik.tu-darmstadt.de/jenkins/job/WebAnno%20(GitHub)%20(master)/de.tudarmstadt.ukp.clarin.webanno$webanno-doc/doclinks/3/#repair_RemoveZeroSizeTokensAndSentencesRepair):

You can temporarily add this to your settings.properties file (mind using the latest build of WebAnno):

```
debug.casDoctor.checks=AllFeatureStructuresIndexedCheck,LinksReachableThroughChainsCheck,FeatureAttachedSpanAnnotationsTrulyAttachedCheck,NoZeroSizeTokensAndSentencesCheck
debug.casDoctor.repairs=RemoveZeroSizeTokensAndSentencesRepair
debug.casDoctor.fatal=true
```

This should try repairing/removing zero-width Tokens from an affected document when you open it. it also checks for all kinds of other potential inconsistencies. I.e. if you were affected by other bugs, you still might get exceptions from CAS Doctor regarding other inconsistencies. In that case, it might be necessary to apply other repair actions as well.

That said, I did not yet fix the "change layer type" action (probably later today). I so far only avoid it creating an inconsistent CAS.

**Comment by reckart** [comment]
Should be fixed now, but commit is filed under #373.
